clang/lib/CAS/IncludeTree.cpp

```diff
@@ -552,7 +860,11 @@ cas::createIncludeTreeFileSystem(IncludeTreeRoot &Root) {
         auto FilenameBlob = File.getFilename();
         if (!FilenameBlob)
           return FilenameBlob.takeError();
-        StringRef Filename = FilenameBlob->getData();
+        SmallString<128> Filename(FilenameBlob->getData());
+        // Strip './' in the filename to match the behaviour of ASTWriter; we
```

Comment:
It will not change the diagnostic paths -- at least not for normal source file diagnostics.  The path in the include-tree is recorded as `./t.c`, which is what is used in the source manager. Only the IncludeTreeFilesystem removes the `./`.  Now, because `status` returns the path as seen by the filesystem, there is a possibility it leaks out without the `./` *somewhere*, but I think normal diagnostics will not change.

I will look at adding a test for this, but if you're okay with it I'll do that as a separate patch.